ModelViewer/ModelViewer/ViewerGraphicsWindow.cpp

```diff
@@ -248,8 +290,62 @@ void ViewerGraphicsWindow::resetView()
     viewportY = 0;
 }
 
+bool ViewerGraphicsWindow::screenshotDialog(const char* format) {
+    if (!initialized) {
+        return false;
+    }
+
+    QString filepath = QFileDialog::getSaveFileName(nullptr,
+        tr("Save screenshot"),
+        QString(),
+        tr("Images (*)"));
+
+    if (!filepath.isEmpty())
+    {
+        ViewerGraphicsWindow::expertFrame(filepath, format);
+    }
+}
+
+bool ViewerGraphicsWindow::saveDialog(QString filePath) {
+    if (!initialized) {
+        return false;
+    }
+
+    QString filepath = QFileDialog::getSaveFileName(nullptr,
+        tr("Save"),
+        QString(),
+        tr("all (*)"));
+
+    if (!filepath.isEmpty())
+    {
+        // TO DO...
+    }
+}
+
+void ViewerGraphicsWindow::expertFrame(QString name, const char* format) {
+    //Initial the image
+    QImage image(width(), height(), QImage::Format_ARGB32);
+    QString filePath = QString("%1.%2").arg(name, format);
+    
+    GLubyte* pixels = (GLubyte*) malloc( 5 * width() * height());
+    if (pixels) {
+        //Read the data from the frame buffer
+        glReadPixels(0, 0, width(), height(), GL_BGRA, GL_UNSIGNED_BYTE, pixels);
+    }
+
+    //Save
+    //To fix: I don't know the reason why the image generated from the framebuffer has a flip,
```

Comment:
This is correct!  OpenGL renders with the +y direction pointing down.  In order for it to look correct, it has to be flipped.